src/update.c

```diff
@@ -367,6 +367,55 @@ static int main_update()
 	retries = 0;
 	timeout = 10;
 
+	/* Get a list of all the iterative manifests from the server_manifest that match
+	* the "from" version, also get a list of all the full manifests in the MoM */
+	struct list *iterative_manifests = NULL;
+	struct list *full_manifests = NULL;
+	struct list *list1;
+	struct file *file1;
+
+	list1 = server_manifest->manifests;
+	while (list1) {
+		file1 = list1->data;
+		list1 = list1->next;
+		if (file1->is_iterative == 1) {
+			/* Discard iterative manifests that are not from the current version */
+			if (file1->from_version == current_version) {
+				iterative_manifests = list_prepend_data(iterative_manifests, file1);
+			}
+		} else {
+			full_manifests = list_prepend_data(full_manifests, file1);
+		}
+	}
+
+	/* Add the full manifest of those bundles that do not have a valid iterative manifest
+	* into the list of iterative_manifests */
+	struct list *list2;
+	struct file *file2;
+	bool add_manifest;
+
+	list1 = full_manifests;
+	while (list1) {
+		add_manifest = true;
+		file1 = list1->data;
+		list1 = list1->next;
+		list2 = iterative_manifests;
+		while (list2) {
+			file2 = list2->data;
+			list2 = list2->next;
+			if (strcmp(file1->bundlename, file2->bundlename) == 0) {
+				add_manifest = false;
+				break;
+			}
+		}
+		if (add_manifest) {
```

Comment:
Drop the `add_manifest` variable and just add it in the `if (strcmp(...) == 0) {}` block above.

Reply:
that won't work, cause we don't want to add the manifest if the strcmp finds a match, we actually want to add it only if there was no match in the whole list, that is why if the add_manifest variable (initialized to true) if it never got set to false then means there was no match and we would add the manifest.

Reply:
ohhhh misread the code, sorry.